modules/wpml/wpml-config.php

```diff
@@ -594,4 +578,136 @@ private function get_field_attribute( SimpleXMLElement $field, $attribute_name )
 
 		return trim( (string) $attributes[ $attribute_name ] );
 	}
+
+	/**
+	 * Returns all wpml-config.xml files in MU plugins.
+	 *
+	 * @since 3.3
+	 *
+	 * @return string[] A context identifier as array key, a file path as array value.
+	 *
+	 * @phpstan-return array<string, string>
+	 */
+	private function get_mu_plugin_files() {
+		if ( ! is_dir( WPMU_PLUGIN_DIR ) ) {
+			return array();
+		}
+
+		$files = array();
+
+		// Search for top level wpml-config.xml file.
+		$file_path = WPMU_PLUGIN_DIR . '/wpml-config.xml';
+
+		if ( file_exists( $file_path ) ) {
+			$files['mu-plugins'] = $file_path;
+		}
+
+		// Search in proxy loaded MU plugins.
+		foreach ( new DirectoryIterator( WPMU_PLUGIN_DIR ) as $file_info ) {
+			if ( $file_info->isDot() || ! $file_info->isDir() ) {
+				continue;
+			}
+
+			$file_path = $file_info->getPathname() . '/wpml-config.xml';
+
+			if ( file_exists( $file_path ) ) {
+				$files[ 'mu-plugins/' . $file_info->getFilename() ] = $file_path;
+			}
+		}
+
+		return $files;
+	}
+
+	/**
+	 * Returns all wpml-config.xml files in plugins.
+	 *
+	 * @since 3.3
+	 *
+	 * @return string[] A context identifier as array key, a file path as array value.
+	 *
+	 * @phpstan-return array<string, string>
+	 */
+	private function get_plugin_files() {
+		$files   = array();
+		$plugins = get_option( 'active_plugins', array() );
+		$plugins = is_array( $plugins ) ? $plugins : array();
+
+		if ( is_multisite() ) {
+			// Don't forget sitewide active plugins thanks to Reactorshop http://wordpress.org/support/topic/polylang-and-yoast-seo-plugin/page/2?replies=38#post-4801829.
+			$sitewide_plugins = get_site_option( 'active_sitewide_plugins', array() );
+
+			if ( ! empty( $sitewide_plugins ) && is_array( $sitewide_plugins ) ) {
+				$plugins = array_merge( $plugins, array_keys( $sitewide_plugins ) );
+			}
+		}
+
+		$plugin_path = trailingslashit( WP_PLUGIN_DIR ) . '%s/wpml-config.xml';
+
+		foreach ( $plugins as $plugin ) {
+			if ( ! is_string( $plugin ) || '' === $plugin ) {
+				continue;
+			}
+
+			$file_dir  = dirname( $plugin );
+			$file_path = sprintf( $plugin_path, $file_dir );
+
+			if ( file_exists( $file_path ) ) {
+				$files[ "plugins/{$file_dir}" ] = $file_path;
+			}
+		}
+
+		return $files;
+	}
+
+	/**
+	 * Returns all wpml-config.xml files in theme and child theme.
+	 *
+	 * @since 3.3
+	 *
+	 * @return string[] A context identifier as array key, a file path as array value.
+	 *
+	 * @phpstan-return array<string, string>
+	 */
+	private function get_theme_files() {
+		$files = array();
+
+		// Theme.
+		$template_path = get_template_directory();
+		$file_path     = "{$template_path}/wpml-config.xml";
+
+		if ( file_exists( $file_path ) ) {
+			$files[ 'themes/' . get_template() ] = $file_path;
+		}
+
+		// Child theme.
+		$stylesheet_path = get_stylesheet_directory();
+		$file_path       = "{$stylesheet_path}/wpml-config.xml";
+
+		if ( $stylesheet_path !== $template_path && file_exists( $file_path ) ) {
```

Comment:
This is probably not the topic of this PR, but I wonder what happens if `wpm-config.php` files from parent and child themes have conflicting data. Is there a hierarchy (like in WordPress) in the way `wpm-config.php` files are loaded?

Reply:
This is done in [28f7c2c](https://github.com/polylang/polylang/pull/1140/commits/28f7c2ccf1c4f16f829c0540a2e77c3b193708a4) and [537bc71](https://github.com/polylang/polylang/pull/1140/commits/537bc715e4f04231b388689b09387daac9bf5022).